target_postgres/sinks.py

```diff
@@ -64,7 +64,7 @@ def process_batch(self, context: dict) -> None:
             context: Stream partition or context dictionary.
         """
         # Use one connection so we do this all in a single transaction
-        with self.connector._connect() as connection:
+        with self.connector._connect() as connection, connection.begin():
```

Comment:
Good choice with the "Begin Once" Transaction management style re https://docs.sqlalchemy.org/en/20/core/connections.html#begin-once